### PR TITLE
Make generate normals optional

### PR DESCRIPTION
**Pull request by lilleyse**
Also adds a fix when trying to normalize zero vectors.

### CODE REVIEW

**Comment by pjcozzi** [comment]
Update README.md with the new CLI argument.

@shehzan10 do you want to review and merge?

File: lib/generateNormals.js [hunk 1]
```diff
@@ -128,7 +129,11 @@ function generateFaceNormals(gltf, primitive, positionSemantic) {
                     Cartesian3.subtract(positionThree, positionOne, scratchNormalTwo);
                     var normal = new Cartesian3();
                     Cartesian3.cross(scratchNormalOne, scratchNormalTwo, normal);
-                    Cartesian3.normalize(normal, normal);
+                    if (Cartesian3.equalsEpsilon(normal, Cartesian3.ZERO, CesiumMath.EPSILON7)) {
+                        Cartesian3.clone(Cartesian3.UNIT_X, normal);
+                    } else {
+                        Cartesian3.normalize(normal, normal);
+                    }
```
Comment (shehzan10) [inline lib/generateNormals.js]: In https://github.com/AnalyticalGraphicsInc/cesium/pull/5010, we decided to go for a stage for degenerating these triangles. I think we should make an issue for that and not use this code for consistency.

File: specs/lib/generateNormalsSpec.js [hunk 2]
```diff
@@ -42,11 +42,11 @@ describe('generateNormals', function(){
...
     });
 
-    it('generates hard normals if they do not exist', function() {
+    it('generates face normals if they do not exist', function() {
         var gltf = gltfNoNormals;
         var byteLengthBefore = 168;
         generateNormals(gltf, {
-            hardNormals : true
+            faceNormals : true
         });
```
Comment (shehzan10) [inline specs/lib/generateNormalsSpec.js]: Can you add a test for smooth normals too?

Comment (lilleyse) [inline specs/lib/generateNormalsSpec.js]: I clarified the wording in the other test, smooth normals is the default so it will be tested there.

**Comment by lilleyse** [comment]
Updated.

File: lib/generateNormals.js [hunk 3]
```diff
@@ -3,6 +3,7 @@ var Cesium = require('cesium');
 var clone = require('clone');
 
 var Cartesian3 = Cesium.Cartesian3;
+var CesiumMath = Cesium.Math;
```
Comment (shehzan10) [inline lib/generateNormals.js]: This can be removed as well (from jsHint failure).

Comment (lilleyse) [inline lib/generateNormals.js]: Ah thanks, fixed now.

**Comment by lilleyse** [comment]
This is ready. (Forgot to bump earlier)

**Comment by shehzan10** [comment]
The README needs to be updated with the `smoothNormals` option right.

**Comment by lilleyse** [comment]
The README.md has it already, but I did add a note to `CHANGES.md`

**Pull request merged**